test/unittests/compile_test.ts

```diff
@@ -0,0 +1,38 @@
+import assert from 'assert';
+import fs from 'fs';
+import path from 'path';
+
+import { exec } from '@actions/exec';
+
+const projectRoot = path.resolve('.');
+
+describe('compile test', () => {
+  afterEach(() => {
+    process.chdir(projectRoot);
+    fs.rmSync('dist', { recursive: true, force: true });
+  });
+
+  it('npm package should compile successfully', async function() {
+    // eslint-disable-next-line @typescript-eslint/no-invalid-this
+    this.timeout(30000); // allow a 30s timeout
+    let execError;
+    try {
+      process.chdir(projectRoot);
+      fs.rmSync('dist', { recursive: true, force: true });
+      let stdout = '';
+      let stderr = '';
+      await exec('npx ttsc --declaration', [], {
+        listeners: {
+          stdout: (data) => stdout += data.toString(), // suppress
+          stderr: (data) => stderr += data.toString(),
```

Comment:
It might make more sense to include the full output on error rather than just the error message? Something like append both to the same string, but if it's stderr set an error flag to true?